controllers/output.go

```diff
@@ -14,13 +14,15 @@ const (
 	coreAPIVersion = "v1"
 )
 
-func (r *HierarchyConfigReconciler) createOutputObject(output konfigmanagerv1.Output, config pkg.Config, resources []pkg.Resource) error {
+func (r *KonfigReconciler) createOutputObject(output konfigmanagerv1.Output, config pkg.Config, resources []pkg.Resource) error {
 	properties := make(map[string]string)
 	if output.Type == "file" || output.Type == "File" {
 		if output.Key == "" {
 			output.Key = "application.properties"
 		}
 		properties[output.Key] = config.GeneratePropertiesFile(resources)
+	} else {
+		properties = config.GetPropertiesMap(resources)
```

Comment:
Should this not be within the config map case?

Reply:
No... For both the cases we first would need the properties map `map[string]string` if its Secret we just convert the each value to `[]byte` the encoding occurs by k8s api

Reply:
My suggestion then would be to have a function that handles the Secret and ConfigMap cases and a function that handles the File and then nest the the function in the `else` block. Feel like the function is getting large.

Reply:
Refactored in two functions